src/MongoDB.Driver/Linq/Processors/ParameterExpressionCollector.cs

```diff
@@ -0,0 +1,48 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace MongoDB.Driver.Linq.Processors
+{
+    /// <summary>
+    /// The expression visitor to collect all parameters in the expression.
+    /// </summary>
+    internal sealed class ParameterExpressionCollector : ExpressionVisitor
```

Comment:
The current implementation of this visitor collects all expression parameters (input and parameters which are used in content), in the previous implementation, we didn't collect input parameters. 
But collecting input expression parameters is not a problem, because we cannot use expression input parameters from parents as input parameters in children, because it's not allowed by syntax restrictions.